vignettes/articles/wrangling_survey_responses.Rmd

```diff
@@ -0,0 +1,233 @@
+---
+title: "Wrangling form submissions"
+output: 
+  html_document:
+    toc: yes
+    toc_depth: 3
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>",
+  message = FALSE
+)
+```
+
+## Introduction
+
+This article demonstrates some techniques for cleaning and wrangling data returned by the `get_submissions` method of the `Kobo` and `Asset` classes. The vignette is broken down by the different question types. In general, users should make heavy use of the question options to give [custom column names](https://support.kobotoolbox.org/question_options.html#data-column-name) and labels for answer options. This will result in better behaving column names and make data cleaning easier.
+
+```{r setup}
+library(kbtbr)
+library(tidyr)
+library(stringr)
+library(dplyr)
+BASE_URL <- "https://kobo.correlaid.org"
+```
+
+```{r include=FALSE}
+kobo <-
+  Kobo$new(base_url_v2 = BASE_URL,
+           kobo_token = Sys.getenv("KBTBR_TOKEN"))
+vcr::use_cassette("kobo-asset-submissions-data", {
+  submissions_df <- kobo$get_submissions("aRo4wg5utWT7dwdnQQEAE7")
+})
+```
+
+
+```{r eval=FALSE}
+kobo <-
+  Kobo$new(base_url_v2 = BASE_URL,
+           kobo_token = Sys.getenv("KBTBR_TOKEN"))
+asset_id <- "aRo4wg5utWT7dwdnQQEAE7"
+submissions_df <- kobo$get_submissions(asset_id)
+```
+
+Let's look at the column names:
+```{r}
+colnames(submissions_df)
+```
+
+The form contains questions of several of the available types in KoboToolbox. The whole list of available types can be found [here](https://support.kobotoolbox.org/question_types.html). More information also on the [xlsform](https://xlsform.org/en/) page. 
+
+Our test form includes:
+
+- select one: `Taking_all_things_to_ould_you_say_you_are` (an example for a bad question label!), `health_self_assessment`, `voluntary_activity`, `trust_in_others`
+- [rating](https://support.kobotoolbox.org/rating_ranking.html#how-to-add-a-rating-question): `trust_in_groups/{category}`, `trust_in_groups/{importance_in_life}`
+- range: `control_over_life`
+- point (location): `locate_hamburg`
+- file upload: `file_upload`
+
+The rest of the columns are metadata automatically added by KoboToolbox. Most of them start with `_`. 
+
+
+
+## Column names 
+### Sorting 
+As could be seen above, the API does not provide a meaningful sorting of the columns.
```

Comment:
Isn't it sorting in the same order as they actually appear in the form? 
I get from line 49  
```
[1] "_id"                                               
 [2] "control_over_life"                                 
 [3] "Taking_all_things_to_ould_you_say_you_are"         
 [4] "voluntary_activity"                                
 [5] "trust_in_groups/Your_family"                       
 [6] "formhub/uuid"                                      
 [7] "_validation_status"                                
 [8] "importance_in_life/Politics"                       
 [9] "_uuid"                                             
[10] "health_self_assessment"                  
```

![image](https://user-images.githubusercontent.com/26052672/132434018-c37bb9c4-dfd4-4cad-8f69-4cec2b084af8.png)

Reply:
mhh i get 


```
#>  [8] "_notes"                                            
#>  [9] "_attachments"                                      
#> [10] "_geolocation"                                      
#> [11] "_status"                                           
#> [12] "__version__"                                       
#> [13] "start"                                             
#> [14] "end"                                               
#> [15] "control_over_life"                                 
#> [16] "Taking_all_things_to_ould_you_say_you_are"         
#> [17] "voluntary_activity"                                
#> [18] "trust_in_groups/Your_family"                       
#> [19] "formhub/uuid"                                      
#> [20] "importance_in_life/Politics"                       
#> [21] "health_self_assessment"                            
#> [22] "importance_in_life/Friends_and_acquaintances"      
#> [23] "trust_in_groups/People_you_know_personally"        
#> [24] "importance_in_life/Family"                         
#> [25] "trust_in_groups/People_you_meet_for_the_first_time"
#> [26] "meta/instanceID"                                   
#> [27] "trust_in_groups/People_of_another_nationality"     
#> [28] "trust_in_groups/People_in_your_neighborhood"       
#> [29] "importance_in_life/Religion"                       
#> [30] "locate_hamburg"                                    
#> [31] "importance_in_life/Leisure_time"           
```

to me this doesn't seem to be sorted in a particular meaningful way (maybe a bit??) .